DashboardClient/DashboardClient.cpp

```diff
@@ -26,6 +26,213 @@ namespace Umati
 			std::shared_ptr<OpcUaTypeReader> pTypeReader)
 			: m_pDashboardDataClient(pDashboardDataClient), m_pPublisher(pPublisher), m_pTypeReader(pTypeReader)
 		{
+			this->startEventThread();
+		}
+
+		DashboardClient::~DashboardClient() {
+			this->stopEventThread();
+		}
+
+		void DashboardClient::startEventThread() {
+			if (m_eventThreadRunning)
+			{
+				LOG(INFO) << "EventThread Running";
+				return;
+			}
+
+			auto func = [this]() {
+				int cnt = 0;
+				while (this->m_eventThreadRunning) {
+					if(!m_eventqueue.empty()) {
+						IDashboardDataClient::StructureChangeEvent sce = m_eventqueue.front();
+						m_eventqueue.pop();
+						this->m_dynamicNodes.push_back(sce.refreshNode);
+						if(sce.nodeAdded || sce.referenceAdded) {
+							this->updateAddDataSet(sce.refreshNode);
+							this->Publish();
+						}
+						if(sce.nodeDeleted || sce.referenceDeleted) {
+							this -> updateDeleteDataSet(sce.refreshNode);
+							if(sce.nodeDeleted == true) {
+							auto search = browsedSimpleNodes.find(sce.refreshNode);
+								if(search != browsedSimpleNodes.end()) {
+									std::shared_ptr<const ModelOpcUa::SimpleNode> simpleNode = search->second;
+									this->deleteAndUnsubscribeNode(*simpleNode);
+								}
+							}
+							this -> Publish();
+    					}
+					}
```

Comment:
Indentation looks suscious.
A `continue` inside should allow to preoces more than 10 Events per seconds (in case of busy OPC UA Servers)

Reply:
Replaced by Threadsafe queue!